### ISSUE TITLE
Add x-ms-discriminator support to OpenAPI for PowerShell

### ISSUE THREAD

**Issue by darrelmiller** [microsoftgraph/microsoft-graph-devx-api#196]
https://github.com/Azure/autorest/blob/master/docs/extensions/readme.md#x-ms-discriminator-value
Should this go in OData.OpenApi reader? Or in OpenAPI.NET v3 -> v2 translation.

Related to: https://github.com/microsoftgraph/msgraph-sdk-powershell/issues/19#

**Comment by darrelmiller** [comment]
Add support for x-ms-discriminator here https://github.com/microsoft/OpenAPI.NET/blob/vnext/src/Microsoft.OpenApi/Models/OpenApiDiscriminator.cs#L49

**Comment by darrelmiller** [comment]
In order to identify derived types we will need to replace all $refs to base types with the following:

```
 oneOf:  
  - $ref: '#/components/schemas/microsoft.graph.directoryObject'
  - $ref: '#/components/schemas/microsoft.graph.user'
  - $ref: '#/components/schemas/microsoft.graph.group'
```

**Comment by darrelmiller** [comment]
More details are explained here https://github.com/microsoftgraph/msgraph-sdk-powershell/issues/19

**Comment by darrelmiller** [comment]
@peombwa  Is the oneOf essential for enabling discriminators or is it just a nice to have?

**Comment by peombwa** [comment]
@darrelmiller It is not essential. We can enable it by having the discriminator object on the base type and `x-ms-discriminator-value` on the derived type. Since the value of the `OData.Type` starts with a `#`, we need to have `x-ms-discriminator-value` on the derived type with its `OData.Type` value. e.g. `#microsoft.graph.user`.

**Comment by peombwa** [comment]
Reopening this so we can investigate the feasibility of using the `DerivedTypeConstraint` that was added to in https://github.com/microsoftgraph/msgraph-metadata/pull/33 to pull in all the derived types of a base type.

**Comment by adhiambovivian** [comment]
@peombwa  do we still need this?

**Comment by maisarissi** [comment]
Yes, we do @adhiambovivian . We need this to be supported so we can have discriminator support in PS SDK.
